### ISSUE TITLE
react/require-default-props is giving a false-positive for no clear reason

### ISSUE THREAD

**Issue by gajus** [yannickcr/eslint-plugin-react#1159]
Plugin v6.10.3.

I am getting an error:

```
/Users/gajus/Documents/dev/applaudience/applaudience-com/src/app/components/Modal/index.js
  7:3  error  propType "+" is not required, but has no corresponding defaultProp declaration  react/require-default-props
```

Though `children` is clearly defined:

```diff
// @flow

import React from 'react';
import './styles.scss';

+type ModalPropsType = {|
+  +children?: React$Element<any>,
+  +onHide: () => void
+|};

export default class extends React.Component {
+  static props: ModalPropsType;

+  static defaultProps = {
+    children: null
+  };

  handleKeypress (event: KeyboardEvent) {
    if (event.key === 'Escape') {
      this.props.onHide();
    }
  }

  componentWillMount () {
    document.addEventListener('keydown', this.handleKeypress.bind(this));
  }

  componentWillUnmount () {
    document.removeEventListener('keydown', this.handleKeypress.bind(this));
  }

  render () {
    return (
      <div styleName='backdrop'>
        <div styleName='modal'>
          <div onClick={this.props.onHide} styleName='close-button' />
          {this.props.children}
        </div>
      </div>
    );
  }
}


```

**Comment by gajus** [comment]
Reading the error more closely, it is looking for "+" property in `defaultProps`, not `children`.

This appears related to https://github.com/yannickcr/eslint-plugin-react/issues/961.

**Comment by gajus** [comment]
@yannickcr can this be assigned a bug label?

**Comment by Dimon70007** [comment]
Hi there. I think, that eslint a good package for development. And will be better to  disable "react/require-default-props" by default, because Eslint should help to develop apps, instead to obstruct.

**Comment by ljharb** [comment]
@Dimon70007 this does help to develop apps - and bad practices (like not providing default props for optional prop types) *need* to be obstructed.

Regardless, this opinion and debate do not belong on this issue, or on this repo.
